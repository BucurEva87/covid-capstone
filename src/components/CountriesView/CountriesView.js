import React from 'react';
import { useSelector } from 'react-redux';
import Country from '../Country/Country';
import style from './CountriesView.module.scss';

const CountriesView = () => {
  const countries = useSelector((state) => state.countries);

  return (
    <section id={style.countries} data-testid="countries-container">
      {
        countries.loading ? (
          <div className={style.loadingContainer}>
            <p>Data is still loading...</p>
            <img src={`${process.env.PUBLIC_URL}images/spinner.gif`} alt="Spinning loader" />
          </div>
        ) : (
          <div id={style.container}>
            {
              countries.list.map((country) => (
                <Country key={country.country} name={country.country} flag={country.flag} />
              ))
            }
          </div>
        )
      }
    </section>
  );
};

export default CountriesView;
