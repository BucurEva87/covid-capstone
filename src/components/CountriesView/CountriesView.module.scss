@import '../../styles/variables';

#countries {
  width: 100%;
}

#container {
  background-color: $primaryColor;
  display: grid;
  grid-template-columns: 1fr 1fr;
  width: 100%;
  text-align: center;
}

article:nth-child(even) {
  background-color: $secondaryColor;
}

.loadingContainer {
  width: 100%;
  text-align: center;
}
