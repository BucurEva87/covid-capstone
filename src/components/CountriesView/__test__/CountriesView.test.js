import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import CountriesView from '../CountriesView';

let getByTestId;

beforeEach(() => {
  const component = render(
    <Provider store={store}>
      <CountriesView />
    </Provider>,
  );

  getByTestId = component.getByTestId;
});

describe('CountriesView component', () => {
  test('renders on the screen', () => {
    expect(getByTestId('countries-container')).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <CountriesView />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
