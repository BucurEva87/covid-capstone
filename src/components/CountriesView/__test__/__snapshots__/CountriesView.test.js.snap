// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountriesView component matches snapshot 1`] = `
<section
  data-testid="countries-container"
  id="countries"
>
  <div
    id="container"
  />
</section>
`;
